Support calls to overloaded Python functions

Calls to stdlib functions with an overloaded type can fail. Example: `int('1')` (when using real stubs for builtins) generates `AssertionError: <class 'mypy.types.Overloaded'> unsupported`.

--- a/mypyc/genops.py
+++ b/mypyc/genops.py
@@ -32,7 +32,7 @@ from mypy.nodes import (
 import mypy.nodes
 from mypy.types import (
     Type, Instance, CallableType, NoneTyp, TupleType, UnionType, AnyType, TypeVarType, PartialType,
-    TypeType, FunctionLike,
+    TypeType, FunctionLike, Overloaded
 )
 from mypy.visitor import NodeVisitor
 from mypy.subtypes import is_named_instance
@@ -239,6 +239,8 @@ class Mapper:
         elif isinstance(typ, PartialType):
             assert typ.var.type is not None
             return self.type_to_rtype(typ.var.type)
+        elif isinstance(typ, Overloaded):
+            return object_rprimitive
         assert False, '%s unsupported' % type(typ)
 
     def fdef_to_sig(self, fdef: FuncDef) -> FuncSignature:

--- a/test-data/genops-basic.test
+++ b/test-data/genops-basic.test
@@ -1660,3 +1660,31 @@ L0:
     r14 = cast(None, r13)
     r15 = None
     return r15
+
+[case testCallOverloaded]
+import m
+def f() -> str:
+    return m.f(1)
+[file m.pyi]
+from typing import overload
+@overload
+def f(x: int) -> str: ...
+@overload
+def f(x: str) -> int: ...
+[out]
+def f():
+    r0 :: object
+    r1 :: str
+    r2 :: object
+    r3 :: int
+    r4, r5 :: object
+    r6 :: str
+L0:
+    r0 = m.module :: static
+    r1 = unicode_0 :: static  ('f')
+    r2 = getattr r0, r1
+    r3 = 1
+    r4 = box(int, r3)
+    r5 = py_call(r2, r4)
+    r6 = cast(str, r5)
+    return r6

